### PR TITLE
 Change NetworkState to read the module configurations from file 

### PR DESCRIPTION
**Pull request by moCello**
ModuleConfig used to get the configurations from the Schedule struct,
which had the hard-coded configurations. Schedule is now depricated
and ModuleConfig reads its configurations from a toml file.
This is to make the configurations of the VM more transparent and
adjusting them easier.
Also change the *InstrumentalizationError* to *InstrumentationError*.

Resolves: #221

### CODE REVIEW

**Review by ureeves — CHANGES_REQUESTED** [review]
Given these changes I think it would make sense to get rid of `NetworkState::new`. We can keep the `Default` since that would make a user think the default configuration is being used, but `new` seems misleading.

**Comment by miloszm** [comment]
> Given these changes I think it would make sense to get rid of `NetworkState::new`. We can keep the `Default` since that would make a user think the default configuration is being used, but `new` seems misleading.

Agree - I also had doubts about it. Will make the appropriate change.

**Comment by miloszm** [comment]
> > Given these changes I think it would make sense to get rid of `NetworkState::new`. We can keep the `Default` since that would make a user think the default configuration is being used, but `new` seems misleading.
> 
> Agree - I also had doubts about it. Will make the appropriate change.

Change is now done.

**Review by ureeves — CHANGES_REQUESTED** [review]
Some naming and type nits.

File: src/state.rs [hunk 1]
```diff
@@ -47,9 +50,26 @@ impl Canon for NetworkState {
 }
 
 impl NetworkState {
-    /// Returns a new empty [`NetworkState`].
-    pub fn new() -> Self {
-        Self::default()
+    /// Returns a [`NetworkState`] based on a specific configuration file
+    pub fn with_config_file(
+        file_path: Option<String>,
```
[the remark this answers is not in the document]
Comment (miloszm) [reply src/state.rs]: Done

File: src/modules.rs [hunk 2]
```diff
@@ -81,66 +87,101 @@ pub enum InstrumentationError {
...
-
-    pub fn with_forbidden_floats(&mut self) -> &mut Self {
-        self.has_forbidden_floats = true;
-        self
-    }
-
-    pub fn with_metering(&mut self) -> &mut Self {
-        self.has_metering = true;
-        self
+    pub fn from_file(file_path: Option<String>) -> Result<Self, VMError> {
```
[the remark this answers is not in the document]
Comment (miloszm) [reply src/modules.rs]: Done

**Review by ZER0 — APPROVED** [review]
LGTM, just a minor nit.
But I was wondering if the `module_config` should now belong to `Contracts` struct instead of `NetworkState`.

**Pull request merged**